WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java

```diff
@@ -198,13 +159,87 @@ public void onErrorResponse(VolleyError error) {
+    private void fetchCategories() {
+        // TODO: Replace with FluxC (GET_CATEGORIES + TaxonomyStore.getCategoriesForSite())
+        WordPress.getRestClientUtilsV1_1().getCategories(mSite.getSiteId(),
+                new RestRequest.Listener() {
+                    @Override
+                    public void onResponse(JSONObject response) {
+                        AppLog.v(AppLog.T.API, "Received site Categories");
+                        credentialsVerified(true);
+
+                        CategoryModel[] models = deserializeCategoryRestResponse(response);
+                        if (models == null) return;
+
+                        SiteSettingsTable.saveCategories(models);
+                        mRemoteSettings.categories = models;
+                        mSettings.categories = models;
+                        notifyUpdatedOnUiThread(null);
+                    }
+                }, new RestRequest.ErrorListener() {
+                    @Override
+                    public void onErrorResponse(VolleyError error) {
+                        AppLog.d(AppLog.T.API, "Error fetching WP.com categories:" + error);
+                    }
+                });
+    }
+
+    private void fetchJetpackProtectAndSsoSettings() {
+        WordPress.getRestClientUtilsV1_1().getJetpackSettings(mSite.getSiteId(), new RestRequest.Listener() {
+            @Override
+            public void onResponse(JSONObject response) {
+                final JSONObject data = response.optJSONObject("data");
+
+                if (data == null) {
+                    AppLog.w(AppLog.T.API, "Unexpected state: Received empty Jetpack settings response");
+                    return;
+                }
+
+                AppLog.v(AppLog.T.API, "Received Jetpack settings response");
+
+                mRemoteJpSettings.monitorActive = data.optBoolean("monitor", false);
+                mRemoteJpSettings.jetpackProtectEnabled = data.optBoolean("protect", false);
+                mRemoteJpSettings.ssoActive = data.optBoolean("sso", false);
+                mRemoteJpSettings.ssoMatchEmail = data.optBoolean("jetpack_sso_match_by_email", false);
+                mRemoteJpSettings.ssoRequireTwoFactor = data.optBoolean("jetpack_sso_require_two_step", false);
+
+                JSONObject jetpackProtectWhitelist = data.optJSONObject("jetpack_protect_global_whitelist");
+                if (jetpackProtectWhitelist != null) {
+                    JSONArray whitelistItems = jetpackProtectWhitelist.optJSONArray("local");
+                    if (whitelistItems != null) {
+                        for (int i = 0; i < whitelistItems.length(); ++i) {
+                            String item = whitelistItems.optString(i, "");
+                            if (!item.isEmpty() && !mRemoteJpSettings.jetpackProtectWhitelist.contains(item)) {
+                                mRemoteJpSettings.jetpackProtectWhitelist.add(item);
```

Comment:
Shouldn't we clear the `mRemoteJpSettings.jetpackProtectWhitelist` before all this?